package.json

```
@@ -66,22 +66,22 @@
     ]
   },
   "devDependencies": {
-    "@typescript-eslint/parser": "^5.45.0",
+    "@typescript-eslint/parser": "^7.10.0",
     "doctoc": "^2.2.1",
     "eslint": "^8.28.0",
-    "eslint-config-prettier": "^8.5.0",
+    "eslint-config-prettier": "^9.1.0",
     "eslint-doc-generator": "^1.7.1",
-    "eslint-plugin-eslint-plugin": "^4.4.1",
-    "eslint-plugin-jest": "^26.9.0",
-    "eslint-plugin-node": "^11.1.0",
-    "eslint-plugin-prettier": "^4.2.1",
-    "globals": "^14.0.0",
-    "husky": "^7.0.4",
-    "jest": "^28.1.3",
-    "lint-staged": "^12.5.0",
-    "npm-run-all": "^4.1.5",
-    "prettier": "^2.7.1",
-    "typescript": "^4.9.3"
+    "eslint-plugin-eslint-plugin": "^6.1.0",
+    "eslint-plugin-jest": "^28.5.0",
+    "eslint-plugin-n": "^17.7.0",
+    "eslint-plugin-prettier": "^5.1.3",
+    "globals": "^15.3.0",
+    "husky": "^9.0.11",
+    "jest": "^29.7.0",
+    "lint-staged": "^15.2.5",
+    "npm-run-all2": "^6.2.0",
+    "prettier": "^3.2.5",
+    "typescript": "^5.4.5"
```

```suggestion
    "typescript": "~5.4.5"
```

Typescript can be breaking in minor versions